code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -176,26 +150,25 @@ paths:
           $ref: '#/components/responses/500'
         '503':
           $ref: '#/components/responses/503'
- 
-  /appInstance/{appInstanceId}:
     delete:
       tags:
-        - Application Instance
-      summary: Terminate an Application Instance
-      description: Terminate a running instance of an application within an Edge Cloud Node
-      operationId: deleteAppInstance
+        - Application
+      summary: Delete an Application from an OP
+      description: Delete all the information and content related to an Application
+      operationId: deleteApp
       parameters:
-        - name: appInstanceId
+        - name: appId
           in: path
-          description: Identificator of the specific application instance that will be terminated
+          description: Identificator of the application to be deleted provided by the OP NBI once the submission was successful
           required: true
           schema:
-            $ref: "#/components/schemas/AppInstanceId"
+            $ref: "#/components/schemas/AppId"
+ 
       responses:
         '202':
-          description: Request accepted to be processed. It applies for async deletion process
+          description: Request accepted
```

Comment:
202 was adopted as @gunjald suggested. And we think is the best option as in CAMARA commonalities indicates that for async process 202 must be showed.
![image](https://github.com/camaraproject/EdgeCloud/assets/150795747/0d9099f8-c7be-499e-bdf4-351eeb82cc30)

We know that there is no callback and for this MVP the way to know when a process is completed is using the GET method. In next releases callbacks should be added.

Reply:
Sorry I should have emphasized it more, but I would expect App APIs (which just manage definitions, i.e. database entries) to finish immediately. Do we expect App APIs to be long-running? In our case, the App APIs never modify instantiated objects (clusters/pods/running containers/etc). They just change the definition of the app, which is just a write to the database. In that case we should use 200, since 202 should only be used for long-running APIs.

In our case, we treat Apps like container images. You can update the container image (in the repository) but it doesn't affect any running containers. If you want to apply the change, you'll need to have the instance (re)pull the image from the repository and redeploy it. We keep a revision ID on the App definition (must like container images use tags/sha256 sums) to track the "version" of the App. The revision ID is incremented whenever the App definition is updated. We do not allow the App to be deleted if the are instances of the App running.

Reply:
App deletion with an app instance running is not allowed too. 

I see your point but correct me if I'm wrong, the deletion process means that the image in the repository will be deleted and maybe this process could take some time (repository could be on a centralized node or shared with other MNOs/CSP) but If you think this is not the case and nobody else thinks otherwise I'm agree to change the response to `200`

Reply:
@javierlozallu Actually our LCM APIs and the Federation EWBI APIs have separate APIs for handling binary images, so deleting the App definition wouldn't actually trigger deleting the referenced image. This is primarily because multiple App definitions may want to share the same binary image(s), and especially for VM images, we want to avoid as much as possible having to re-upload large binary images if the user is re-creating an App definition for example.

Reply:
/app APIs in my opinion could be subjective on how a platform implements the handling. Some of the platform may just store in a central repository all the information from the API or some could even send it to the environments where they may expect it to launch later only on those environments. So if we provide a GET call to retrieve status of the /app API i think we can accommodate more wider platform implementations.